catalog/src/test/java/org/killbill/billing/catalog/dao/TestCatalogOverrideUsageTierSqlDao.java

```diff
@@ -0,0 +1,130 @@
+                assertEquals(rehydrated.getTargetUsageDefRecordId(), obj1.getTargetUsageDefRecordId());
+                return null;
+            }
+        });
+    }
+
+    @Test(groups = "slow")
+    public void testGetTargetUsageDefinition() throws Exception {
+
+        final CatalogOverrideUsageTierModelDao obj1 = new CatalogOverrideUsageTierModelDao((short) 1, 2L, 3L);
+        final CatalogOverrideUsageTierModelDao obj2 = new CatalogOverrideUsageTierModelDao((short) 2, 5L, 3L);
+        final CatalogOverrideUsageTierModelDao obj3 = new CatalogOverrideUsageTierModelDao((short) 4, 7L, 3L);
+        final CatalogOverrideUsageTierModelDao nobj1 = new CatalogOverrideUsageTierModelDao((short) 4, 7L, 4L);
+
+        performTestInTransaction(new WithCatalogOverrideUsageTierSqlDaoTransaction<Void>() {
+            @Override
+            public Void doTransaction(final CatalogOverrideUsageTierSqlDao sqlDao) {
+                sqlDao.create(obj1, internalCallContext);
+                sqlDao.create(obj2, internalCallContext);
+                sqlDao.create(obj3, internalCallContext);
+                sqlDao.create(nobj1, internalCallContext);
+
+                final List<String> keys = new ArrayList<String>();
+                keys.add("1,2");
+                keys.add("2,5");
+                keys.add("4,7");
+                final List<Long> targetUsages = sqlDao.getTargetUsageDefinition(keys, keys.size(), internalCallContext);
+                assertEquals(targetUsages.size(), 1);
+                assertEquals(targetUsages.get(0), new Long(3));
+                return null;
+            }
+        });
+    }
+
+    @Test(groups = "slow")
+    public void testGetTargetUsageDefWithSameTierOverrideAndDifferentUsagePriceOverride() throws Exception {
+
+        final CatalogOverrideUsageTierModelDao obj1 = new CatalogOverrideUsageTierModelDao((short) 1, 2L, 3L);
+        final CatalogOverrideUsageTierModelDao obj2 = new CatalogOverrideUsageTierModelDao((short) 1, 2L, 4L);
+
+        performTestInTransaction(new WithCatalogOverrideUsageTierSqlDaoTransaction<Void>() {
+            @Override
+            public Void doTransaction(final CatalogOverrideUsageTierSqlDao sqlDao) {
+                sqlDao.create(obj1, internalCallContext);
+                sqlDao.create(obj2, internalCallContext);
+
+                final List<String> keys = new ArrayList<String>();
+                keys.add("1,2");
+                final List<Long> targetUsages = sqlDao.getTargetUsageDefinition(keys, keys.size(), internalCallContext);
+                assertEquals(targetUsages.size(), 2);
+                assertEquals(targetUsages.get(0), new Long(3));
+                assertEquals(targetUsages.get(1), new Long(4));
+                return null;
+            }
+        });
+    }
+
+    private interface WithCatalogOverrideUsageTierSqlDaoTransaction<T> {
+
+        public <T> T doTransaction(final CatalogOverrideUsageTierSqlDao sqlDao);
```

Comment:
Could these internal classes be extracted in a base class somehow? Something like:

```
TestCatalogOverrideTierDefinitionSqlDao extends TestCatalogOverrideBaseSqlDao
TestCatalogOverrideUsageDefinitionSqlDao extends TestCatalogOverrideBaseSqlDao
TestCatalogOverrideUsageTierSqlDao extends TestCatalogOverrideBaseSqlDao
[...]
TestCatalogOverrideBaseSqlDao extends CatalogTestSuiteWithEmbeddedDB
```

where `TestCatalogOverrideBaseSqlDao` would have a generic `WithCatalogOverrideSqlDaoTransaction<T,U>` and `performTestInTransaction(final WithCatalogOverrideSqlDaoTransaction<T, U> callback)`?

If it's too hard, no big deal, but it would simplify a bit the test setup.